src/main/java/com/wanted/babdoduk/user/domain/entity/User.java

```diff
@@ -0,0 +1,50 @@
+package com.wanted.babdoduk.user.domain.entity;
+
+import com.wanted.babdoduk.common.domain.entity.BaseTimeEntity;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Table(name = "users")
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
```

Comment:
접근제한이 `PROTECTED` 가 아닌 `PRIVATE` 인 이유가 있을까요?

Reply:
Empty Constructor는 JPA가 런타임에 [Entity 객체 인스턴스화, 프록시 객체 생성 등에 사용]하기 위해 필요하지만, 개발자가 직접 Empty Constructor를 사용할 수는 없도록 접근 제어자에 제한을 둔다고 알고 있습니다. 저는 그동안 `private`로 줄곧 정의해온 편이였는데, 만약 같은 package나 상속받는 Entity에서 연관관계가 정의될 수도 있을 상황을 고려한다면 `protected`로 두는 게 나을 수도 있겠네요. 😀

[Entity 객체 인스턴스화, 프록시 객체 생성 등에 사용]: https://www.baeldung.com/jpa-no-argument-constructor-entity-class

Reply:
`PRIVATE` 이였을 때 혹시 경고메시지 나오지 않았나요? Entity 생성의 경우 내부적으로 Proxy를 사용하기 때문에 접근 제한은`PUBLIC` or `PROTECTED` 로 해야하는 것으로 알고 있는데.. 혹시 제가 놓친 부분이 있을까요?

Reply:
애플리케이션을 실행했을 때는 에러 메시지가 따로 출력되지는 않아서 테스트를 진행해 봤습니다.

```java
@Entity
@Table(name = "dogs")
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class Dog {

    @Id
    @GeneratedValue
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
}
```

그리고 인스턴스 하나를 삽입해서 조회해 봤는데 Error가 발생했습니다. 정말로 런타임에 말씀해주신 예외가 발생했네요.

```text
2023-11-02T01:26:51.084+09:00 ERROR 61548 --- [nio-8000-exec-1] o.h.p.p.bytebuddy.ByteBuddyProxyFactory  : HHH000143: Bytecode enhancement failed because no public, protected or package-private default constructor was found for entity: com.wanted.babdoduk.user.domain.entity.User. Private constructors don't work with runtime proxies

java.lang.NoSuchMethodException: com.wanted.babdoduk.user.domain.entity.User$HibernateProxy$Ve6wNOy7.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3617) ~[na:na]
	at java.base/java.lang.Class.getConstructor(Class.java:2303) ~[na:na]
```

저는 그동안 Entity 내에서 연관관계를 웬만하면 두지 않는 쪽으로 작성을 했어서 아예 이런 이슈가 발생할 가능성이 있는지를 생각하지 않았었는데, 꽤 중대할 수 있었던 에러였던 것 같습니다... 내부 로직이 어떻게 돌아가는지도 알 수 있어야 한다는 게 무슨 의미인지 와닿는 것 같습니다. 이슈 체크해주신 점 감사드립니다!!